sonic_data_client/db_client.go

```diff
@@ -161,21 +171,39 @@ func (c *DbClient) StreamRun(q *queue.PriorityQueue, stop chan struct{}, w *sync
 	c.q = q
 	c.channel = stop
 
-	for gnmiPath, tblPaths := range c.pathG2S {
-		if tblPaths[0].field != "" {
+	if subscribe.GetSubscription() == nil {
+		log.V(2).Infof("No incoming subscription, it is considered a dialout connection.")
+		// NOTE: per https://github.com/Azure/sonic-telemetry/blob/master/doc/dialout.md#dialout_client_cli-and-dialout_server_cli
+		// TELEMETRY_CLIENT subscription doesn't specificy type of the stream.
+		// Handling it as a ON_CHANGE stream for backward compatibility.
+		// TODO: We need to decide wheter this should be ON_CHANGE or SAMPLE
+		for gnmiPath := range c.pathG2S {
 			c.w.Add(1)
 			c.synced.Add(1)
-			if len(tblPaths) > 1 {
-				go dbFieldMultiSubscribe(gnmiPath, c)
+			go streamOnChangeSubscription(c, gnmiPath)
+		}
+	} else {
+		log.V(2).Infof("Stream subscription request received, mode: %v, subscription count: %v",
+			subscribe.GetMode(),
+			len(subscribe.GetSubscription()))
+
+		for _, sub := range subscribe.GetSubscription() {
+			log.V(2).Infof("Sub mode: %v, path: %v", sub.GetMode(), sub.GetPath())
+			subMode := sub.GetMode()
+
+			if subMode == gnmipb.SubscriptionMode_SAMPLE {
+				c.w.Add(1)
+				c.synced.Add(1)
+				go streamSampleSubscription(c, sub, subscribe.GetUpdatesOnly())
+			} else if subMode == gnmipb.SubscriptionMode_ON_CHANGE {
+				c.w.Add(1)
+				c.synced.Add(1)
```

Comment:
could you please add some comment about values added to data structures?

Reply:
Ack. Some of these (e.g. `c.w` and `c.synced` ) are existing components. I might have omitted documenting those due because of that.